src/transformers/generation/utils.py

```diff
@@ -1431,8 +1427,6 @@ def generate(
         batch_size = inputs_tensor.shape[0]
 
         # 4. Define other model kwargs
-        model_kwargs["output_attentions"] = generation_config.output_attentions
-        model_kwargs["output_hidden_states"] = generation_config.output_hidden_states
```

Comment:
Instead of pulling information from `generation_config` to pass around through `model_kwargs`, let's use `generation_config` directly. 

A single object to hold all generation parameterization.